### DISCUSSION TITLE
Proper way to import lodash?

### DISCUSSION THREAD

**Discussion by kennerstross** [nuxt/nuxt#17917] [Questions] [answered]
I've been using lodash with Nuxt 3 with no problems like this...

`import _ from 'lodash'
`
But now that I've started using jest, too, that style doesn't work. Jest wants...

`import * as _ from 'lodash'
`
... but that doesn't work with Nuxt 3. Is there a way I can make both of them happy?

**Comment by fallsimply** [comment, marked as answer]
I think this is an issue with CommonJS vs ModuleJS. Jest is a CommonJS tool and Nuxt v3 is ModuleJS based.
There is a new testing tool called [Vitest](https://vitest.dev/guide/#adding-vitest-to-your-project), which uses the same build tool as Nuxt v3. 
Heres [the Migration Guide from Jest](https://vitest.dev/guide/migration.html#migrating-from-jest)

**Reply by kennerstross** [reply]
Thank you fallsimply - I'll give Vitest a try!

**Comment by wobsoriano** [comment]
There's a module for lodash if you're interested

https://github.com/cipami/nuxt-lodash

**Reply by kennerstross** [reply]
Cool. Thanks, wobsoriano!

**Reply by benkissi** [reply]
the project is archived

**Comment by AloisSeckar** [comment]
Since this topic got ressurected:

Proper way of importing `lodash` in 2025 is **not to import it** and to use https://es-toolkit.dev/ instead. Or at least `lodash-es`, if you really insist.

Both because slowly dropping support for CJS solutions and also because of `lodash` immunity to be tree-shaken during build. Last but not least because the project is dead for almost 6 years already which is like eterninty in the current web development.
